app/models/alchemy/page/page_naming.rb

```diff
@@ -69,24 +57,24 @@ def visible_ancestors
 
     private
 
-    def update_descendants_urlnames
-      self.reload
-      descendants.each do |descendant|
-        next if descendant.redirects_to_external?
-        descendant.update_urlname!
-      end
+    # Sets the urlname attribute to a url friendly slug.
+    # NOTE: Does not save
+    def set_urlname
+      write_attribute(:urlname, nested_urlname)
     end
 
-    # Sets the urlname to a url friendly slug.
-    # Either from name, or if present, from urlname.
-    # If url_nesting is enabled the urlname contains the whole path.
-    def set_urlname
-      if Config.get(:url_nesting)
-        value = slug
-      else
-        value = urlname
-      end
-      self[:urlname] = nested_url_name(value)
+    # Returns the urlname to a url friendly slug.
+    #
+    def nested_urlname
+      url_name = [
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `url_name`. Did you mean `urlname`? (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)